qa/os/src/test/java/org/elasticsearch/packaging/test/EnrollmentProcessTests.java

```diff
@@ -68,6 +71,40 @@ public void test10AutoFormCluster() throws Exception {
         assertThat(makeRequest("https://localhost:9200/_cluster/health"), containsString("\"number_of_nodes\":2"));
     }
 
+    public void test20DockerAutoFormCluster() throws Exception {
+        assumeTrue("only docker", distribution.isDocker());
+        // First node
+        installation = runContainer(distribution(), builder().envVar("ELASTIC_PASSWORD", "password"));
+        verifyContainerInstallation(installation);
+        verifySecurityAutoConfigured(installation);
+        waitForElasticsearch(installation);
+        assertThat(makeRequestAsElastic("https://localhost:9200/_cluster/health", "password"), containsString("\"number_of_nodes\":1"));
+        final String node1ContainerId = Docker.getContainerId();
+        Shell.Result createTokenResult = installation.executables().createEnrollmentToken.run("-s node");
+        assertThat(Strings.isNullOrEmpty(createTokenResult.stdout), is(false));
+        final List<String> noWarningOutputLines = Arrays.stream(createTokenResult.stdout.split("\\n"))
+            .filter(line -> line.startsWith("WARNING:") == false)
+            .collect(Collectors.toList());
+        assertThat(noWarningOutputLines.size(), equalTo(1));
+        // installation refers to second node from now on
+        installation = runAdditionalContainer(
+            distribution(),
+            builder().envVar("ENROLLMENT_TOKEN", noWarningOutputLines.get(0)).extraArgs("--publish 9301:9300 --publish 9201:9200")
+        );
+        // TODO Make our packaging test methods aware of multiple installations, see https://github.com/elastic/elasticsearch/issues/79688
+        waitForElasticsearch(installation);
+        verifyContainerInstallation(installation);
+        verifySecurityAutoConfigured(installation);
+        // Allow some time for the second node to join the cluster, we can probably do this more elegantly in
+        // https://github.com/elastic/elasticsearch/issues/79688
+        Thread.sleep(15000);
+        // verify that the two nodes formed a cluster
+        assertThat(makeRequestAsElastic("https://localhost:9200/_cluster/health", "password"), containsString("\"number_of_nodes\":2"));
```

Comment:
You could use `PackagingTestCase.assertBusy()`:

```suggestion
        // Allow some time for the second node to join the cluster
        assertBusy(() ->
            // verify that the two nodes formed a cluster
            assertThat(makeRequestAsElastic("https://localhost:9200/_cluster/health", "password"), containsString("\"number_of_nodes\":2")));
```

Reply:
Thanks, I hadn't seen that,will change !